scripts/validate_data.py

```diff
@@ -1,74 +1,79 @@
-""" Python script to validate data
-
-Run as:
-
-    python3 scripts/validate_data.py data
-"""
-
 from pathlib import Path
 import sys
 import hashlib
 
-
 def file_hash(filename):
-    """ Get byte contents of file `filename`, return SHA1 hash
-
+    """ Get SHA1 hash of the contents of a file
     Parameters
     ----------
     filename : str
         Name of file to read
-
     Returns
     -------
     hash : str
         SHA1 hexadecimal hash string for contents of `filename`.
     """
-    # Open the file, read contents as bytes.
-    # Calculate, return SHA1 has on the bytes from the file.
-    # This is a placeholder, replace it to write your solution.
-    raise NotImplementedError(
-        'This is just a template -- you are expected to code this.')
-
+    sha1 = hashlib.sha1()
+    with open(filename, 'rb') as fobj:
+        while True:
+            data = fobj.read(65536)  # Read in 64k chunks
+            if not data:
+                break
+            sha1.update(data)
+    return sha1.hexdigest()
 
 def validate_data(data_directory):
     """ Read ``data_hashes.txt`` file in `data_directory`, check hashes
-
     Parameters
     ----------
     data_directory : str
         Directory containing data and ``data_hashes.txt`` file.
-
     Returns
     -------
     None
-
     Raises
     ------
     ValueError:
         If hash value for any file is different from hash value recorded in
         ``data_hashes.txt`` file.
     """
-    # Read lines from ``data_hashes.txt`` file.
-    # Split into SHA1 hash and filename
-    # Calculate actual hash for given filename.
-    # If hash for filename is not the same as the one in the file, raise
-    # ValueError
-    # This is a placeholder, replace it to write your solution.
-    raise NotImplementedError('This is just a template -- you are expected to code this.')
+    data_path = Path(data_directory)
+    data_hashes_file = data_path / 'hash_list.txt'
 
+    if not data_hashes_file.is_file():
```

Comment:
It's a bit of a judgement call here whether you'd do the explicit check or let the code error below.

Reply:
Thanks for the review..